src/Compilers/Core/MSBuildTask/CopyRefAssembly.cs

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Reflection.Metadata;
+using System.Reflection.PortableExecutable;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.CodeAnalysis.BuildTasks
+{
+    /// <summary>
+    /// If the source and destination ref assemblies are different,
+    /// this task copies the source over to the destination.
+    /// </summary>
+    public class CopyRefAssembly : Task
+    {
+        [Required]
+        public string SourcePath { get; set; }
+
+        [Output]
+        [Required]
+        public string DestinationPath { get; set; }
+
+        public CopyRefAssembly()
+        {
+            TaskResources = ErrorString.ResourceManager;
+        }
+
+        public override bool Execute()
+        {
+            if (!File.Exists(SourcePath))
+            {
+                Log.LogErrorWithCodeFromResources("General_ExpectedFileMissing", SourcePath);
+                return false;
+            }
+
+            if (!File.Exists(DestinationPath))
+            {
+                return Copy();
+            }
+
+            var source = ExtractMvid(SourcePath);
+            var destination = ExtractMvid(DestinationPath);
```

Comment:
Need to consider the case where ExtractMvid fails.  Consider that the developer corrupts a file on disk, it's held with a non-sharing lock, or a simple text file is accidentally put into the location.  This version of the code would cause the CopyAssemblyRef task to fail.  

I think the design should be that we keep the old one as an optimization.  Any error trying to implement that optimization should fall back to the non-optimized Copy path (with a log message).